src/main/resources/liquibase/crop_changelog/15_4_1.xml

```diff
@@ -173,6 +173,37 @@
 		</sql>
 	</changeSet>
 
+	<changeSet id="v15.4.1-10" author="aldrin">
+		<preConditions onFail="MARK_RAN">
+			<not>
+				<sqlCheck expectedResult="0">
+					-- Check if there are stocks that don't have GROUPGID stock property.
+					SELECT COUNT(*) FROM stock s
```

Comment:
The precondition query can be improved by:
1. not performing a join to `germplsm` table
2. doing a `NOT EXISTS `instead of `LEFT JOIN` to `stockprop` table
            `AND NOT EXISTS (select 1 from stockprop sp WHERE sp.stock_id = s.stock_id AND sp.type_id = 8330);`